src/client/LiveConf.js

```diff
@@ -501,68 +492,58 @@ class LiveConf {
-        conf.live_recording.recording_config.recording_params.xc_params.video_bitrate = LadderTemplate[720].bit_rate;
-        conf.live_recording.recording_config.recording_params.xc_params.enc_height = 720;
-        conf.live_recording.recording_config.recording_params.xc_params.enc_width = 1280;
-        break;
-      case 540:
-        conf.live_recording.recording_config.recording_params.ladder_specs.unshift(
-          LadderTemplate[540],
-          LadderTemplate["540_low"]
-        );
-        conf.live_recording.recording_config.recording_params.xc_params.video_bitrate = LadderTemplate[540].bit_rate;
-        conf.live_recording.recording_config.recording_params.xc_params.enc_height = 540;
-        conf.live_recording.recording_config.recording_params.xc_params.enc_width = 960;
-        break;
-      default:
-        throw new Error("ERROR: Probed stream does not conform to one of the following built in resolution ladders [4096, 2160], [1920, 1080] [1280, 720], [960, 540]");
-    }
+    conf.live_recording.recording_config.recording_params.xc_params.enc_height = videoStream.height;
+    conf.live_recording.recording_config.recording_params.xc_params.enc_width = videoStream.width;
 
+    // Determine video recording bitrate and ABR ladder
+    let topLadderRate = 0
+    for (let i = 0; i < ladderProfile.video.length; i ++) {
+      let elem = ladderProfile.video[i];
+      if (elem.height > videoStream.height)
+        continue;
+      if (elem.bit_rate > topLadderRate) {
+        topLadderRate = elem.bit_rate
+      }
+      elem.media_type = 1;
+      elem.stream_name = "video";
+      elem.stream_index = 0;
+      elem.representation = "videovideo_" + videoStream.width + "x" + videoStream.height + "@" + elem.bit_rate;
+      conf.live_recording.recording_config.recording_params.ladder_specs.push(elem);
+    }
+    // Currently the recording bitrate is the top bitrate of the ladder (it will be configurable separately in the future)
+    conf.live_recording.recording_config.recording_params.xc_params.video_bitrate = topLadderRate;
 
+    // Determine audio recording bitrate and ABR ladder
     let globalAudioBitrate = 0;
     let nAudio = 0;
 
     for(let i = 0; i < Object.keys(audioStreams).length; i ++ ) {
-      let audioLadderSpec = {...LadderSpecAudio};
+      let audioLadderSpec = {};
       const audioIndex = Object.keys(audioStreams)[i];
       const audio = audioStreams[audioIndex];
-      audioLadderSpec.bit_rate = audio.recordingBitrate;
+
+      // Find ladder rung for the specific channel layout (2 or 6 channels)
+      for (let i = 0; i < ladderProfile.audio.length; i ++) {
+        let elem = ladderProfile.video[i];
+        if (elem.channels == audio.recordingChannels) {
+          audioLadderSpec = elem;
+          break;
+        }
+      }
+      if (audioLadderSpec == {}) {
+        // If no channels layout match, just use the first element in the ladder
+        audioLadderSpec = ladderProfile.video[0];
```

Comment:
Did you mean to use `ladderProfile.audio[0];` here?

Reply:
Fixed!
Also fixed the 'codecs' which was overwritten here by mistake.